eth/client.go

```diff
@@ -883,84 +870,28 @@ func (c *client) ContractAddresses() map[string]ethcommon.Address {
 }
 
 func (c *client) CheckTx(tx *types.Transaction) error {
-	ctx, cancel := context.WithTimeout(context.Background(), c.txTimeout)
-	defer cancel()
-
-	receipt, err := bind.WaitMined(ctx, c.backend, tx)
-	if err != nil {
-		return err
-	}
-
-	if receipt.Status == uint64(0) {
-		return fmt.Errorf("tx %v failed", tx.Hash().Hex())
-	} else {
-		return nil
+	receipts := make(chan *transactionReceipt, 10)
+	txSub := c.tm.Subscribe(receipts)
+	defer txSub.Unsubscribe()
+
+	for {
+		select {
+		case err := <-txSub.Err():
+			return err
+		case receipt := <-receipts:
+			if tx.Hash() == receipt.TxHash {
+				if receipt.err != nil {
+					return receipt.err
+				}
+				if receipt.Status == uint64(0) {
+					return fmt.Errorf("transaction failed txHash=%v", receipt.TxHash)
```

Comment:
```suggestion
					return fmt.Errorf("transaction failed txHash=%v", receipt.TxHash.Hex())
```

I think we need to call Hex() in order to log a hex string here?